test/core.vwtest.json

```diff
@@ -4221,5 +4221,70 @@
       "stdout": "train-sets/ref/truncated_model_crash.stdout"
     },
     "bash_command": "./negative-test.sh {VW} --no_stdin --cb_explore_adf --cover 3 --noconstant --quiet -i model-sets/truncated_model"
+  },
+  {
+    "id": 357,
+    "desc": "bad argument in one_of string",
+    "diff_files": {
+      "stderr": "train-sets/ref/cb_bad_one_of_string.stderr"
+    },
+    "bash_command": "./negative-test.sh {VW} --cb_adf -d train-sets/cb_test.ldf --cb_type bad_type"
+  },
+  {
+    "id": 358,
+    "desc": "bad argument in one_of int",
+    "diff_files": {
+      "stderr": "train-sets/ref/cb_bad_one_of_int.stderr"
+    },
+    "bash_command": "./negative-test.sh {VW} -k -d train-sets/lda-2pass-hang.dat --lda 10 -c --passes 2 --holdout_off --math-mode 5"
+  },
+  {
+    "id": 359,
+    "desc": "--probabilities --oaa one-indexed",
+    "diff_files": {
+      "stderr": "train-sets/ref/oaa_one_ind_probabilities.stderr",
+      "oaa_one_ind_probabilities.predict": "pred-sets/ref/oaa_one_ind_probabilities.predict"
+    },
+    "vw_command": "-d train-sets/probabilities_one_ind.dat --probabilities --oaa=4 --loss_function=logistic -p oaa_one_ind_probabilities.predict",
+    "input_files": [
+      "train-sets/probabilities_one_ind.dat"
+    ]
+  },
+  {
+    "id": 360,
+    "desc": "--probabilities --oaa zero-indexed",
+    "vw_command": "-d train-sets/probabilities_zero_ind.dat --probabilities --oaa=4 --loss_function=logistic -p oaa_zero_ind_probabilities.predict --indexing 0",
+    "diff_files": {
+      "stderr": "train-sets/ref/oaa_zero_ind_probabilities.stderr",
+      "oaa_zero_ind_probabilities.predict": "pred-sets/ref/oaa_zero_ind_probabilities.predict"
+    },
+    "input_files": [
+      "train-sets/probabilities_zero_ind.dat"
+    ]
+  },
+  {
+    "id": 361,
+    "desc": "--probabilities --oaa zero-indexed without flag",
+    "vw_command": "-d train-sets/probabilities_zero_ind.dat --probabilities --oaa=4 --loss_function=logistic -p oaa_zero_ind_no_flag_probabilities.predict",
+    "diff_files": {
+      "stderr": "train-sets/ref/oaa_zero_ind_no_flag_probabilities.stderr",
+      "oaa_zero_ind_no_flag_probabilities.predict": "pred-sets/ref/oaa_zero_ind_no_flag_probabilities.predict"
+    },
+    "input_files": [
+      "train-sets/probabilities_zero_ind.dat"
+    ]
+  },
+  {
+    "id": 362,
+    "desc": "--probabilities --oaa mixed-indexing",
+    "vw_command": "-d train-sets/probabilities_mixed_ind.dat --probabilities --oaa=4 --loss_function=logistic -p oaa_mixed_probabilities.predict",
+    "diff_files": {
+      "stderr": "train-sets/ref/oaa_mixed_probabilities.stderr",
+      "stdout": "train-sets/ref/oaa_mixed_probabilities.stdout",
+      "oaa_mixed_probabilities.predict": "pred-sets/ref/oaa_mixed_probabilities.predict"
```

Comment:
nit: It may be easier to rename this field to "predict" then extrapolate the file name from the value

Reply:
The key of this mapping is the filename in the test dir, that is the file it searches for to compare to the ref file which is the value. stderr and stdout are special values. Everything else is just a file name

Reply:
Yes I meant we could use the ref file to extrapolate the file we search for, assuming they both have the same name

Reply:
I feel like it is easier to be explicit here, otherwise we have some implicit mapping between the filename produced by the test and the file it is being compared to